### DISCUSSION TITLE
color coded workspaces and containers

### DISCUSSION THREAD

**Discussion by giaaaacomo** [zen-browser/desktop#2630] [Ideas]
hi, i often end up opening tabs in the wrong workspace and container. while it's easy to move tabs between workspaces, having set up a default container for the workspace means that opening the tab in the wrong workspace requires a new tab to be loaded of the same page and often a new login to occur. while this is not something that can be fixed, having color codes or other identifications systems would surely help, for example border color for workspace and fill color for container, or vice versa, or maybe icons next to the title. another thing that could help would be changing color of the text and icons so that they reflect the color assigned to the workspace, meaning that for example the +, ⇄, and sidebar button to open a new tab would be of the same color of the current workspace

**Comment by kwilcz** [comment]
There already is color code for containers, and workspaces can have their separate window theme.

**Reply by giaaaacomo** [reply]
> There already is color code for containers, and workspaces can have their separate window theme.

could you show me the settings for it then? i see "Use workspace colors as primary color" but not a way to define the workspace's color

**Reply by kwilcz** [reply]
![image](https://github.com/user-attachments/assets/2773066a-3cbe-4e30-aaa0-1bad2c9776fc)
For workspaces, just `RMB on toolbar` -> `Change theme colors`

**Reply by giaaaacomo** [reply]
> For workspaces, just `RMB on toolbar` -> `Change theme colors`

thanks for that. i totally missed the `change theme colors` on that menu. this btw is a wrong ux choice, the right way to do this should be in the workspace properties (where you can set the icon) and in the settings

**Reply by d0x7** [reply]
> > For workspaces, just `RMB on toolbar` -> `Change theme colors`
> 
> thanks for that. i totally missed the `change theme colors` on that menu. this btw is a wrong ux choice, the right way to do this should be in the workspace properties (where you can set the icon) and in the settings

Oh wow, didn't see that either. But that honestly is just a bad UI, why would you put it there instead of on a right click on the workspaces?

**Reply by Dehelssey** [reply]
it's really too hidden... especially when you selected the "single toolbar" look & feel :eyes:

**Reply by rnhurt** [reply]
This changes the color of the theme but not the color of the container.

**Reply by Dehelssey** [reply]
the color of the container is set within the container option (where you can pick the icon as well) 
![image](https://github.com/user-attachments/assets/96ab3535-faca-456c-bb48-1139bde96cff)

**Reply by rnhurt** [reply]
Yes, but that only changes the color of the text (name) of the container in the URL bar, AFAIK.  I'm looking to change the color of the entire theme based on container color.  Or at least make containers a bit more recognizable.

**Reply by Dehelssey** [reply]
Ooooh. OK. Not sure you can change the whole browser's color based on the container of a tab.

You can either have a workspace that opens every tab in the container of your choice by defaut and then color code each workspace to match the container.

Otherwise there's a Zen mod that gives the tabs a bold border that matches their container: https://zen-browser.app/mods/3ff55ba7-4690-4f74-96a8-9e4416685e4e/

I'm using it it's pretty efficient

**Reply by rnhurt** [reply]
Thank you for the pointer to the "Colored container tab" mod.  I gave it a try and it does highlight the current tab, but it removes the highlight from all the other tabs also.  This actually makes the problem worse for me.  

In my daily workflow I'm logged into multiple AWS accounts simultaneously, often on the same console page.  Since I have multiple tabs on the same page with different accounts the only way to tell them apart is by their color.  :frowning:

**Reply by Dehelssey** [reply]
Arg, sorry it didn't do the job for you. :(

**Comment by rnhurt** [comment]
I'm using the [AWS SSO Containers](Better Active Tab) Firefox extension to automatically "containerize" some of my tabs.  It updates the color indicator on the tab and the container name in the URL bar, but that's not quite enough for me.  I'm looking for a way to change the theme of the whole theme to the container color.

The problem I'm trying to solve is forcing me to know which container I'm in.  For example, I want the theme to be BRIGHT RED when I'm in a "production" AWS account.  Or having the theme be soft green when I'm a "playground" account.  This way I know when it's safe to make changes and when I should really, really be paying attention.

The [Containers Theme](https://addons.mozilla.org/en-US/firefox/addon/containers-theme/) Firefox extension looks like it might do what I want but it appears to not know about Zen browser peculiarities.  I even looked into the source to see if I could tweak it a bit but I don't know what JS to use to change the theme in Zen browser.
